changelog/11871.feature.rst

```diff
@@ -0,0 +1,4 @@
+Added support for reading ``file_or_dir`` positional arguments from a file
+using the prefix character '@', like e.g.:
+
+``pytest @tests.txt``
```

Comment:
```suggestion
Added support for reading positional arguments from a file using the prefix character '@', like e.g.: ``pytest @tests.txt``
```

Reply:
You're right, it is not only positional arguments. `-k foo` in the file is treated as a `-k`-argument. Is this ok, as long as I update the changelog and warn about it? Not sure if and how it would be possible to avoid it.